torchgeo/datasets/geo.py

```diff
@@ -339,7 +339,8 @@ def __init__(
         """Initialize a new Dataset instance.
 
         Args:
-            root: root directory where dataset can be found
+            root: root directory or list of absolute filepaths where
+                dataset can be found
```

Comment:
Renamed to paths. As you will see from mypy, a lot of datasets implements e.g. `_verify` that expect it to be a (single) directory. Would it be possible to generalise this method and put it in RasterDataset? And then handle paths instead of root.

Reply:
`_verify` methods generally seem to do:

If no file match `self.filename_glob` in root:
-> Do extract archive
If no archive:
-> Do download

And some check_integrity / checksum.

I think I have mostly updated these to allow `self.paths` to be a list, but introducing method `list_files`. The only "breaking" change is that archives are now extracted to the same dir as the archive, not root. This possible do not break stuff though?

I think these methods can be generalised, but this should be a separate issue/pr.